File: src/types/movie.ts
export interface Movie {
    id: number | string; 
    title: string;
    release_date: string;
    overview: string;
    poster_path: string;
    backdrop_path?: string;
    genre_ids?: number[];
    video: boolean,
    vote_average: number;
    vote_count?: number;
    original_language?: string;
}

export interface ReviewAuthorDetails {
    name: string;
    username: string;
    avatar_path: string | null;
    rating: number | null;
}

export interface Review {
    author: string;
    author_details: ReviewAuthorDetails;
    content: string;
    created_at: string;
    id: string;
    updated_at: string;
    url: string;
}

export interface MovieQueryResult {
    title: string;
    year: number;
}

File: backend/app/models/movie.py
# app/models.py
from pydantic import BaseModel
from typing import Optional, List

class Movie(BaseModel):
    id: int
    title: str
    overview: Optional[str] = None
    release_date: Optional[str] = None
    poster_path: Optional[str] = None
    backdrop_path: Optional[str] = None
    vote_average: Optional[float] = None
    vote_count: Optional[int] = None
    popularity: Optional[float] = None
    genre_ids: List[int] = []
    adult: bool = False
    original_language: Optional[str] = None
    original_title: Optional[str] = None

class RecommendMoviesRequest(BaseModel):
    prompt: str

class MovieRecommendation(BaseModel):
    title: str
    year: Optional[int] = None
    reason: Optional[str] = None

class HealthCheck(BaseModel):
    status: str
    message: str
    version: str
    tmdb_configured: bool
    openai_configured: bool